### PR TITLE
Adding ppc64le architecture support on travis-ci

### PR DESCRIPTION
**Pull request by kishorkunal-raj**
Hi,
I had added ppc64le(Linux on Power) architecture support on Travis-CI in the PR and looks like its been successfully added. I believe it is ready for the final review and merge. The Travis-CI build logs can be verified from the link below.
https://travis-ci.com/github/kishorkunal-raj/python-musicbrainzngs/builds/205851077

Reason behind running tests on ppc64le: This package is included in the ppc64le versions of RHEL and Ubuntu - this allows the top of tree to be tested continuously as it is for Intel, making it easier to catch any possible regressions on ppc64le before the distros begin their clones and builds. This reduces the work in integrating this package into future versions of RHEL/Ubuntu.

Please have a look.

Regards,
Kishor Kunal Raj

### CODE REVIEW

**Comment by alastair** [comment]
Hi,
Thanks for the contribution here. This project is a pure python package, with no additional calls to other compiled code, so I'm not sure if it's strictly necessary to have a specific ppc version of the tests.
Do you have a link to information about the RHEL or Ubuntu build process that shows that they use the results of upstream CI to verify results before packaging?

**Comment by kishorkunal-raj** [comment]
Hi @alastair,
Thanks for your reply. I understand this is a pure python package and thinks @seth-priya or @gerrith3 can answer to your question regarding link to information.

**Comment by gerrith3** [comment]
@alastair regarding links to Ubuntu and RHEL build processes:  I don't think RHEL releases their build processes, although for fedora pretty much everything should be out in the open.  IBM and Canonical are parterns and we are evolving and optimizing the process that they use to do builds - this is a part of that process to ultimately simplify builds for other architectures across the board.  As such, there isn't a lot of data on the future processes, e.g. the 2021 and 2022 builds - you are right at the beginnings of the process optimisation here.

**Comment by alastair** [comment]
Hi @gerrith3, thanks so much for the comment. Just to clarify, I have no problem merging this PR, I was just interested in knowing how we were found, and how this change can help downstream distros.

I guess this will use more travis minutes as will run all tests twice, is that right? This isn't a huge issue for us, as it's such a low-traffic project, and we probably don't even come close to the limits.
Does Github Actions allow for different architectures yet?

From my original question - do you foresee any reason why a pure python package might see different test results on ppc and x64? My intuition is that any bugs are almost certain to be because of python interpreter bugs on that arch rather than any code in the project, but I'd be happy to be shown that this is useful for downstream packages too!

I'm curious to know your process for selecting projects to submit these PRs to, I assume you're just looking for package upstreams that also have a travis configuration option. Is that right?

**Comment by alastair** [comment]
@kishorkunal-raj as per our [contribution guide](https://github.com/alastair/python-musicbrainzngs/blob/master/CONTRIBUTING.md) can you please sign your commit? thanks!

**Comment by gerrith3** [comment]
@alastair you were "found" by my team because all of musicbrainz is in the ppc64le distro for Ubuntu 20.04, as an example:

```
ubuntu@gerrit:~$ sudo apt-cache search musicbrainz
grilo-plugins-0.3-base - Framework for discovering and browsing media - Plugins
abcde - A Better CD Encoder
audiotools - Collection of audio handling programs for the command line
beets - music tagger and library organizer
beets-doc - music tagger and library organizer - documentation
clementine - modern music player and library organizer
flactag - Tagger for whole-album FLAC files using data from MusicBrainz
glyrc - command-line interface to libglyr
goobox - CD player and ripper with GNOME 3 integration
isrcsubmit - extract ISRCs from audio CDs and submit them to MusicBrainz
kid3 - KDE audio tag editor
kid3-cli - Command line audio tag editor
kid3-core - Audio tag editor core libraries and data
kid3-qt - Audio tag editor
libcoverart-dev - library to access the Cover Art Archive (development files)
libcoverart-doc - library to access the Cover Art Archive (developer documentation)
libcoverart1 - library to access the Cover Art Archive (C wrapper functions)
libcoverartcc1v5 - library to access the Cover Art Archive
libdiscid-dev - library for creating MusicBrainz DiscIDs (development files)
libdiscid-doc - library for creating MusicBrainz DiscIDs (documentation)
libdiscid0 - library for creating MusicBrainz DiscIDs
libghc-musicbrainz-dev - interface to MusicBrainz XML2/JSON web services
libghc-musicbrainz-doc - interface to MusicBrainz XML2/JSON web services; documentation
libghc-musicbrainz-prof - interface to MusicBrainz XML2/JSON web services; profiling libraries
libglyr-dev - search engine for music-related metadata - development files
libglyr-doc - search engine for music-related metadata - documentation
libglyr1 - search engine for music-related metadata
libmusicbrainz-discid-perl - Perl interface to the MusicBrainz libdiscid library
libmusicbrainz5-2 - Library to access the MusicBrainz.org database (C wrapper functions)
libmusicbrainz5-dev - Library to access the MusicBrainz.org database (development files)
libmusicbrainz5-doc - Library to access the MusicBrainz.org database (developer documentation)
libmusicbrainz5cc2v5 - Library to access the MusicBrainz.org database
libwebservice-musicbrainz-perl - XML based Web service API to the MusicBrainz database
picard - Next-Generation MusicBrainz audio files tagger
python-libdiscid-doc - libdiscid binding for Python (documentation)
python-musicbrainzngs-doc - Documentation for the Python Musicbrainz NGS interface modules
python3-libdiscid - libdiscid binding for Python 3
python3-libdiscid-dbg - libdiscid binding for Python 3 (debug extension)
python3-musicbrainzngs - Python 3 bindings for the Musicbrainz NGS webservice interface
ripit - Textbased audio CD ripper
ubuntu@gerrit:~$ uname -a
Linux gerrit 5.4.0-54-generic #60-Ubuntu SMP Fri Nov 6 10:37:33 UTC 2020 ppc64le ppc64le ppc64le GNU/Linux

```
Pure python packages definitely exist - however we often (maybe 15-20% of the time)  find that people *think* their package is architecture independent, but then there is some dependency, some missing component, some version of python, some library, etc. that wasn't ported to ppc64le, or maybe it was ported but your dependency references an x86-64 only component.  

Basically, the best way to guarantee that something works is to actually test it on the target environment, which is why we have set up the full Travis ppc6le ecosystem for open source to use.

BTW, you mentioned credits - if you need more for travis, just put in a request at support.travis-ci.com for additional credits - at about 10 credits per minute, so if you needed another 5,000 minutes, you'd ask for 50k credits there, just as an example.  That shouldn't be an issue, and if it is tag me and I'll help.

**Comment by kishorkunal-raj** [comment]
> @kishorkunal-raj as per our [contribution guide](https://github.com/alastair/python-musicbrainzngs/blob/master/CONTRIBUTING.md) can you please sign your commit? thanks!

Hi @alastair ,
I had signed the commit. Please take a look

**Comment by alastair** [comment]
I've switched to github actions, so this PR is no longer relevant
